src/x/serialize/decoder.go

```diff
@@ -35,6 +35,7 @@ var (
 )
 
 type decoder struct {
+	checked.FinalizeableOnce
```

Comment:
Perhaps embed in the serialize/encoder too?

Reply:
Done.